librad/src/net/gossip.rs

```diff
@@ -64,16 +66,19 @@ pub use storage::*;
 pub use types::*;
 
 #[derive(Clone)]
-pub enum ProtocolEvent {
-    SendAdhoc(Box<Hello>),
-    Connect(Box<Hello>),
+pub enum ProtocolEvent<A> {
+    SendAdhoc(Box<Hello<A>>),
+    Connect(Box<Hello<A>>),
     Disconnect(PeerId),
 }
 
+// Nb.: this type exists only because clippy otherwise complains that
```

Comment:
I don't understand why it's better to have this `Box` indirection -- isn't `Hello` already a box of sorts?

Reply:
> isn't `Hello` already a box of sorts

I'm not sure they're equivalent. `Box` is a pointer to heap memory right? So it's the size of a pointer. Whereas `Hello` is the size of `PeerInfo` + `Rpc` (give or take). So `Box`ing it would reduce the size to align on, _I think_.

Reply:
Hm ya. I'll leave it like this for now, perhaps some bestestest practices / idiomaticness folks can explain how this is supposed to be done at some point